### ISSUE TITLE
Throw Import Errors Delayed

### ISSUE THREAD

**Issue by osanj** [osanj/lava#13]
Right now if the import the import of `lava` fails an exception is thrown right away. It would be nicer to rather throw on the last moment possible, i.e. if the user calls `lava.instance()` or `lava.devices()` and the initialization has failed previously.

_Motivation:_
During "standard" continuous integration graphics cards and Vulkan might be not available. The import errors that are currently being thrown break these pipelines. The only option right now is to have ugly manual handling of `import lava`.

**Comment by osanj** [comment]
PR #14

**Comment by osanj** [comment]
The current implementation only solved the issue for importing `lava`. When importing a specific module errors still occur:
```python
# vulkan is not available
import lava as lv  # does not blow up
from lava.buffer import BufferCPU # does blow up
```

**Comment by osanj** [comment]
Fix and test added with https://github.com/osanj/lava/commit/68b647ea8837dda45b59160830bfacb997dfede8. Will be released as `0.3.1`.
